moveit_ros/planning_interface/moveit_cpp/src/planning_component.cpp

```diff
@@ -126,12 +126,14 @@ const std::string& PlanningComponent::getName() const
   return group_name_;
 }
 
-bool PlanningComponent::plan(const PlanRequestParameters& parameters)
+PlanningComponent::PlanSolution PlanningComponent::plan(const PlanRequestParameters& parameters)
 {
+  last_plan_solution_.reset(new PlanSolution());
   if (!joint_model_group_)
   {
     ROS_ERROR_NAMED(LOGNAME, "Failed to retrieve joint model group for name '%s'.", group_name_.c_str());
-    return false;
+    last_plan_solution_->error_code = MoveItErrorCode(moveit_msgs::MoveItErrorCodes::FAILURE);
+    return *last_plan_solution_;
```

Comment:
I think we should return a copy instead of a reference to the internal object like this. It's way too easy ending up either with a dangling reference or an undefined trajectory.

Reply:
Did I miss something .? `PlanningComponent::PlanSolution` as return type returns a copy, not a reference, right .?